src/transformers/models/clip/processing_clip.py

```diff
@@ -109,5 +126,13 @@ def decode(self, *args, **kwargs):
     @property
     def model_input_names(self):
         tokenizer_input_names = self.tokenizer.model_input_names
-        feature_extractor_input_names = self.feature_extractor.model_input_names
-        return list(dict.fromkeys(tokenizer_input_names + feature_extractor_input_names))
+        image_processor_input_names = self.image_processor.model_input_names
+        return list(dict.fromkeys(tokenizer_input_names + image_processor_input_names))
+
+    @property
+    def feature_extractor_class(self):
+        warnings.warn(
+            "`feature_extractor_class` is deprecated and will be removed in v5. Use `image_processor_class` instead.",
+            FutureWarning,
+        )
+        return self.image_processor_class
```

Comment:
Should we also allow to give `feature_extractor` ?

Reply:
I mean adding a  property `feature_extractor` too ?

Reply:
Yes, that's a good point. I'll add that too.

Reply:
I've added a `feature_extractor` class to all the processors. Let me know if that matches what you were thinking :)